### ISSUE TITLE
Integrate Enketo Express API into ODK Central

### ISSUE THREAD

**Issue by MartijnR** [enketo/enketo-express#97]
This is not actually an issue for Enketo Express, but I'm putting it here so we can track it better in our Github kanban DIAL project board.

It entails essentially adding the same kind of Enketo integration into ODK Central that KoBoToolbox and Ona enjoy.

https://forum.opendatakit.org/t/integration-of-enketo-into-odk-central/13992

**Comment by jnm** [comment]
* Yes, we can [set a global base path](https://github.com/kobotoolbox/enketo-express/issues/81) to avoid setting up a separate hostname _!_
* Yes, we can configure Enketo with [environment variables](https://github.com/enketo/enketo-express/blob/master/config/sample.env) _!_

**Comment by jnm** [comment]
Quick note: I got Enketo running using `base path`, sharing a hostname with ODK Central. However, because Enketo passes through cookies from the original request to the OpenRosa endpoint, ODK Central fails us for trying to use two authentication methods simultaneously:
https://github.com/getodk/central-backend/blob/f6502a04fa3f0e167df1b9c0d04b4775ecd75baa/lib/http/preprocessors.js#L120-L131

Failing request from Enketo to Central looks like:
```
HEAD /v1/key/aCthSZgrdBdob!KKoNwqIM!Rw4F7!5g5gu7LfEFfzWYkAw$mIx0cawdh6iAHY7eg/projects/1/formList?formID=widgets HTTP/1.0
X-Forwarded-Proto: https
Host: service:8383
Connection: close
cookie: __Host-session=SWcvwj8Mle$0bCnGuIZ9Wf3QLZjX21DKLAosCvXQ2$Q59tL5eyS0LZMjwHYb00L3; __enketo_meta_deviceid=s%3Aenketo%3AHR622ed005SBa6sv.yGlV73OCvXD9FXGEN2oI1WDSlNzrI6%2BU8LlWgEPUnZo; _csrf=023lQLydaLmTWKnXJ9QsYAPu; __enketo_=s%3AeyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoiZHVyciIsInBhc3MiOiJvayJ9.C9ii409_ntdaPvhEyDPNcvzM9WLxW-oEuE8aM2pEP_A.T0jcLLsY7sbkt9I9ryc7JjReyhsENAFdRsfIFV9N064; __enketo_logout=true; __enketo_meta_username=s%3Adurr.b80PALx90Hk5%2FDWQGDK131tswKI82B1d513kC3cywuk
X-OpenRosa-Version: 1.0
Date: Mon, 04 May 2020 05:57:07 GMT
```

The auth that we want to use is the key in the URL itself.

**Comment by lognaturel** [comment]
[Central v0.9](https://forum.getodk.org/t/odk-central-v0-9/28296) was just released and leverages Enketo for form previews! 🎉 Thanks for the help making that a reality, @MartijnR and @jnm. This uses cookie auth for published forms and public links that wrap a Central App User token for drafts.

The next step is to enable data submission. The current challenge is either satisfying Central's CSRF protection requirement or waiving the requirement just for local Enketo.

**Comment by MartijnR** [comment]
Woohoo!! :party: Congratulations on getting this feature in!

**Comment by lognaturel** [comment]
Data submission is ready to go pending release of https://github.com/enketo/enketo-express/pull/197! I think after that either this issue can be closed or @MartijnR you could use it to track some other wishlist items for Central to add.

**Comment by MartijnR** [comment]
Excellent! Just released 2.3.10 with your fix.
